openedx/features/course_experience/tests/test_url_helpers.py

```diff
@@ -0,0 +1,227 @@
+                ... etc ...
+            }
+        }
+        """
+        super().setUpClass()
+
+        with super().setUpClassAndTestData():
+
+            # Make Split Mongo course.
+            course_run = CourseFactory.create(
+                org='TestX',
+                number='UrlHelpers',
+                run='split',
+                display_name='URL Helpers Test Course',
+                default_store=ModuleStoreEnum.Type.split,
+            )
+            with cls.store.bulk_operations(course_run.id):
+                section = ItemFactory.create(
+                    parent_location=course_run.location,
+                    category='chapter',
+                    display_name="Generated Section",
+                    publish_item=False,
+                )
+                subsection = ItemFactory.create(
+                    parent_location=section.location,
+                    category='sequential',
+                    display_name="Generated Subsection",
+                    publish_item=False,
+                )
+                unit = ItemFactory.create(
+                    parent_location=subsection.location,
+                    category='vertical',
+                    display_name="Generated Unit",
+                    publish_item=False,
+                )
+                component = ItemFactory.create(
+                    parent_location=unit.location,
+                    category='problem',
+                    display_name="Generated Problem Component",
+                    publish_item=False,
+                )
+                cls.store.publish(course_run.location, None)
+
+            # Make (deprecated) Old Mongo course.
+            deprecated_course_run = CourseFactory.create(
+                org='TestX',
+                number='UrlHelpers',
+                run='mongo',
+                display_name='URL Helpers Test Course (Deprecated)',
+                default_store=ModuleStoreEnum.Type.mongo,
+            )
+            with cls.store.bulk_operations(deprecated_course_run.id):
+                deprecated_section = ItemFactory.create(
+                    parent_location=deprecated_course_run.location,
+                    category='chapter',
+                )
+                deprecated_subsection = ItemFactory.create(
+                    parent_location=deprecated_section.location,
+                    category='sequential',
+                    display_name="Generated Subsection",
+                )
+                deprecated_unit = ItemFactory.create(
+                    parent_location=deprecated_subsection.location,
+                    category='vertical',
+                    display_name="Generated Unit",
+                )
+                deprecated_component = ItemFactory.create(
+                    parent_location=deprecated_unit.location,
+                    category='problem',
+                    display_name="Generated Problem Component",
+                )
```

Comment:
One thing I often do when I'm getting these kinds of issues with modulestore-related tests is to strip out the bulk_operations call and do it step by step.

Is there a reason you're publishing with the previous set of item creations, but not publishing with this one?

Reply:
Other things you might want to check:

* If the item you want to pull is missing, what _is_ there before you make that call? Just the root course node? The chapter?
* What do you get if you force it to pull from one branch or the other (draft vs. published)?
* Do your usage keys have version information attached to them?

Reply:
> Is there a reason you're publishing with the previous set of item creations, but not publishing with this one?

The Old Mongo items were working fine without the publish line. I was adding random lines to the Split items to try to get something, anything working.

> If the item you want to pull is missing, what is there before you make that call? Just the root course node? The chapter?

After poking at this for a bit, I got the feeling that _nothing_ was being created in the Split modulestore other than the course run itself. That led me to think that all the other items were being created in Old Mongo, even though their root parent was a Split course.

Wrapping the whole section in a default_store context manager did the trick:
```
        with modulestore().default_store(ModuleStoreEnum.Type.split):
            course_run = CourseFactory.create(
                org='TestX',
                number='UrlHelpers',
                run='split',
                display_name='URL Helpers Test Course',
            )
            with cls.store.bulk_operations(course_run.id):
                section = ItemFactory.create(
                    parent_location=course_run.location,
                    category='chapter',
                    display_name="Generated Section",
                )
                ... etc
```
all good now; thanks for the tips!

Reply:
https://github.com/edx/edx-platform/pull/26835/commits/f0bc0f5a5b12b61afa042ffe68f53bea6aae549d#diff-cb49127ab4b7862f424d68c90a7e4df8260d6f8cdc9a460799eca11f830db2eeR65-R66, if you are curious

Reply:
(after merging, I replaced `with modulestore().default_store` in favor of `with cls.store.default_store`)